openpype/tools/workfiles/window.py

```diff
@@ -134,7 +135,9 @@ def set_context(self, asset_id, task_name, filepath, workfile_doc):
             "<b>Created:</b>",
             creation_time.strftime(datetime_format),
             "<b>Modified:</b>",
-            modification_time.strftime(datetime_format)
+            modification_time.strftime(datetime_format),
+            "<b>User:</b>",
+            pwd.getpwuid(filestat.st_uid).pw_name
```

Comment:
Nice! Isn't pwd UNIX only which means this will fail on Windows? We might need to get a Windows equivalent solution as well?

Reply:
> yes it doesn't work on windows and in fact, it is breaking OpenPype there because it will crash during host initialization on missing pwd module. Getting file ownership on Windows from Python is PITA. This is quick "multiplatform" snipped I've put together:
> 
> ```python
> def get_user_name(file: str) -> str:
>     if platform.system() == "Windows":
>         import win32security
>         sd = win32security.GetFileSecurity(
>             file, win32security.OWNER_SECURITY_INFORMATION)
>         owner_sid = sd.GetSecurityDescriptorOwner()
>         name, domain, type = win32security.LookupAccountSid(None, owner_sid)
>         return f"{name}@{domain}"
>     else:
>         import pwd
>         filestat = os.stat(file)
>         return pwd.getpwuid(filestat.st_uid).pw_name
> ```
> 
> but the Windows part depends on `win32security` that is not standard part of Python and needs to be added to OP dependencies.
> 
> In that case it needs to be added to `pyproject.toml` and `poetry.lock` and this PR should target for `release/next-minor`.

nice, thanks for the snippet! (damn windows..)

I will add this and target `next-minor` instead

Reply:
in the future it would be nice to add a way to filter workfiles by user as well and perhaps introduce a way to query the full user name from the SSO so it's more readable

Reply:
@antirotor done here https://github.com/ynput/OpenPype/pull/5087/commits/499071eda208606379dd3aab4dca3505790ff7e2 ! Tested in Linux and it still works fine, if you could try it on Windows please